### ISSUE TITLE
getting Unsupported block type error while using edw_authorized_views module

### ISSUE THREAD

**Issue by bksprasad** [terraform-google-modules/terraform-google-bigquery#174]
### TL;DR

**Here is Code Snippet:** `module "edw_authorized_views" {
  source  = "terraform-google-modules/bigquery/google//modules/authorization"
  version = "~> 5.4.2"

  dataset_id = module.edw_bq_data_set.bigquery_dataset.dataset_id
  project_id = module.edw_bq_data_set.bigquery_dataset.project

  authorized_views = [
    {
      project_id = data.google_project.edw-prcd-bq.project_id
      dataset_id = "edw_publish"
      table_id   = "dim_employer"
    }
  ]

  authorized_datasets = [
    {
      project_id = data.google_project.edw-stage-bq.project_id
      dataset_id = "edw"
    }
  ]
}`

**Here is the error**

╷
│ Error: Unsupported block type
│ 
│   on .terraform/modules/edw_authorized_views/modules/authorization/main.tf line 68, in resource "google_bigquery_dataset_access" "authorized_dataset":
│   68:   dataset {
│ 
│ Blocks of type "dataset" are not expected here.

### Expected behavior

it should successfully do terra form plan

### Terraform Configuration

```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {

    google = {
      source  = "hashicorp/google"
      version = ">= 3.63, < 5.0"
    }
  }

  provider_meta "google" {
    module_name = "blueprints/terraform/terraform-google-bigquery/v5.4.2"
  }
```

### Terraform Version

```sh
terraform {
  required_version = ">= 0.13"
  required_providers {

    google = {
      source  = "hashicorp/google"
      version = ">= 3.63, < 5.0"
    }
  }

  provider_meta "google" {
    module_name = "blueprints/terraform/terraform-google-bigquery/v5.4.2"
  }
```

**Comment by imrannayer** [comment]
@bksprasad Can you plz execute `terraform --version` to check what version of google provider you are using? This feature was released in version [4.12](https://github.com/hashicorp/terraform-provider-google/blob/main/CHANGELOG.md#4120-february-28-2022)

**Comment by bksprasad** [comment]
Here are the details

bash-5.1$ terraform --version
Terraform v1.0.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/google v4.48.0
+ provider registry.terraform.io/hashicorp/google-beta v4.48.0

Your version of Terraform is out of date! The latest version
is 1.3.8. You can update by downloading from https://www.terraform.io/downloads.html
bash-5.1$

From: Imran Nayer ***@***.***>
Date: Wednesday, February 15, 2023 at 3:51 PM
To: terraform-google-modules/terraform-google-bigquery ***@***.***>
Cc: Siva Prasad ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [terraform-google-modules/terraform-google-bigquery] getting Unsupported block type error while using edw_authorized_views module (Issue #174)

@bksprasad<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbksprasad&data=05%7C01%7Csprasad%40castlighthealth.com%7C5f12a63a8bea496558bf08db0faf83d1%7C89443928ab244de3af1470f6a05cbc8f%7C0%7C0%7C638121018750396094%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=4qOA%2FM%2BKhLkCVKYUB%2FIUIXWgwcuUJMkE8t5aF170Oks%3D&reserved=0> Can you plz execute terraform --version to check what version of google provider you are using? This feature was released in version 4.12<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fhashicorp%2Fterraform-provider-google%2Fblob%2Fmain%2FCHANGELOG.md%234120-february-28-2022&data=05%7C01%7Csprasad%40castlighthealth.com%7C5f12a63a8bea496558bf08db0faf83d1%7C89443928ab244de3af1470f6a05cbc8f%7C0%7C0%7C638121018750396094%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=yNL38B2pyxJRPN%2BuzDdN0YXrZ7TnCidWgPx%2BznxY6yk%3D&reserved=0>

**Comment by imrannayer** [comment]
It seems like you have appropriate version of the provider. I just ran a quick test and it is working fine for me.  Can u add following code in your root module and try `terraform init --upgrade` before executing `terraform plan`

```
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 4.0"
    }
  }
}
```

**Comment by bksprasad** [comment]
Hi Imran,

It not working I have added following code . Terraform plan is giving errors. Here is the error

h-5.1$ terraform plan
╷
│ Error: Unsupported block type
│
│   on main.tf line 1169, in module "vaccineservice_publish_bq_data_set":
│ 1169:   terraform {
│
│ Blocks of type "terraform" are not expected here.
╵
bash-5.1$



module "vaccineservice_bq_data_set" {
  source  = "terraform-google-modules/bigquery/google"
  version = "~> 5.4.2"
  terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 4.0"
    }
  }
}
  dataset_id                  = var.vaccineservice_schema
  dataset_name                = var.vaccineservice_schema
  description                 = "Dataset for all tables coming from vaccineservice schema"
  project_id                  = data.google_project.edw-raw-bq.project_id
  location                    = "us-central1"

  access = concat(
    [
      {
        role          = "OWNER"
        special_group = "projectOwners"
      }
    ],
    var.phi ? [] : [{
      role          = "WRITER"
      group_by_email = var.developer_rbac_group
    }]
  )
}


From: Imran Nayer ***@***.***>
Date: Wednesday, February 15, 2023 at 8:02 PM
To: terraform-google-modules/terraform-google-bigquery ***@***.***>
Cc: Siva Prasad ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [terraform-google-modules/terraform-google-bigquery] getting Unsupported block type error while using edw_authorized_views module (Issue #174)

It seems like you have appropriate version of the provider. I just ran a quick test and it is working fine for me. Can u add following code in your root module and try terraform init --upgrade before executing terraform plan

terraform {

  required_providers {

    google = {

      source  = "hashicorp/google"

      version = "~> 4.0"

    }

    google-beta = {

      source  = "hashicorp/google-beta"

      version = "~> 4.0"

    }

  }

}

**Comment by imrannayer** [comment]
@bksprasad that code is not part of module. Add is separately in your code.

**Comment by bksprasad** [comment]
Hi Imran,


I have commented the previous code and added the new code block in providers.tf. I am getting below error



#provider "google" {
# scopes = [
  #  https://www.googleapis.com/auth/cloud-platform,
   # https://www.googleapis.com/auth/userinfo.email,
  #]
#}

#provider "google-beta" {

#}


terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "~> 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "~> 4.0"
    }
  }
}


Error:


bash-5.1$ terraform init --upgrade
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.
╷
│ Error: Duplicate required providers configuration
│
│   on versions.tf line 2, in terraform:
│    2:   required_providers {
│
│ A module may have only one required providers configuration. The required providers were previously configured at providers.tf:13,3-21.
╵

bash-5.1$
From: Imran Nayer ***@***.***>
Date: Thursday, February 16, 2023 at 6:04 AM
To: terraform-google-modules/terraform-google-bigquery ***@***.***>
Cc: Siva Prasad ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [terraform-google-modules/terraform-google-bigquery] getting Unsupported block type error while using edw_authorized_views module (Issue #174)

@bksprasad<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbksprasad&data=05%7C01%7Csprasad%40castlighthealth.com%7C32b77250b08a40ea10d308db1026aaee%7C89443928ab244de3af1470f6a05cbc8f%7C0%7C0%7C638121530485724984%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=kV9s16OwkSmwlcRvIBIwX86%2FTufd9DUwnD6jRlC6e5E%3D&reserved=0> that code is not part of module. Add is separately in your code.

**Comment by apeabody** [comment]
This should be fixed by https://github.com/terraform-google-modules/terraform-google-bigquery/pull/180
